src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusButton
+
+    enum Size {
+        Small,
+        Large
+    }
+
+    enum Type {
+        Normal,
+        Danger,
+        WithoutBackground,
+        DangerWithoutBackground
```

Comment:
I wonder if `WithoutBackground` and `DangerWithoutBackground` should really be a type here.

The built-in `Button` type already comes with a flag `flat` which kind of does the same thing. 

What do you think about sticking to that "standard" and have a `flat` property instead, or, if that makes things too complex, how about we introduce something like:

```
StatusFlatButton {
  type: Normal // or Danger
}
```

Reply:
I think we can use it like base low level component. But without C++ extensions its always looks ugly. Enums in QML is ugly too :)
So we can "hide" it by "alias" like this:
`
/// StatusDangerButton.qml
       StatusButton {
           type: StatusButton.Type.Danger  
       }
`

Reply:
Right, then how about we introduce `StatusDangerButton` and `StatusFlatButton` in this PR accordingly?